src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -202,9 +226,29 @@ public void doGenerateAllBundles(StaplerRequest req, StaplerResponse rsp) throws
         if (supportPlugin != null) {
             supportPlugin.setExcludedComponents(remove);
         }
+        prepareBundle(rsp, components);
+    }
+
+    /**
+     * Generates a support bundle with only requested components.
+     * @param components component names separated by comma.
+     * @param rsp
+     * @throws IOException
+     */
+    @RequirePOST
+    public void doGenerateBundle(@QueryParameter("components") String components, StaplerResponse rsp) throws IOException {
+        if (components == null) {
+            rsp.sendError(HttpServletResponse.SC_BAD_REQUEST, "components parameter is mandatory");
+            return;
+        }
+        List<String> componentNames = Arrays.asList(components.split(","));
+        logger.fine("Selecting components...");
+        prepareBundle(rsp, getComponents().stream().filter(c -> componentNames.contains(c.getId())).collect(Collectors.toList()));
```

Comment:
It would be a user error but maybe it should check if the list is not empty?

Reply:
Done.